tests/ga/test_monitor.py

```diff
@@ -53,29 +53,36 @@ def _create_monitor_handler(enabled_operations=None, iterations=1):
         * run_and_wait() - invokes run() and wait() on the MonitorHandler
 
     """
-    if enabled_operations is None:
-        enabled_operations = []
-
-    def run(self):
-        if len(enabled_operations) == 0 or self._name in enabled_operations:  # pylint: disable=protected-access
-            run.original_definition(self)
-    run.original_definition = PeriodicOperation.run
-
-    with mock_wire_protocol(DATA_FILE) as protocol:
-        protocol_util = MagicMock()
-        protocol_util.get_protocol = Mock(return_value=protocol)
-        with patch("azurelinuxagent.ga.monitor.get_protocol_util", return_value=protocol_util):
-            with patch.object(PeriodicOperation, "run", side_effect=run, autospec=True):
+    def periodic_operation_run_mock(self):
+        if self._name in enabled_operations:  # pylint: disable=protected-access
+            periodic_operation_run_mock.original_definition(self)
+    periodic_operation_run_mock.original_definition = PeriodicOperation.run
+
+    periodic_operation_run_patch = None
+
+    try:
+        with mock_wire_protocol(DATA_FILE) as protocol:
+            protocol_util = MagicMock()
+            protocol_util.get_protocol = Mock(return_value=protocol)
+            with patch("azurelinuxagent.ga.monitor.get_protocol_util", return_value=protocol_util):
                 with patch("azurelinuxagent.ga.monitor.MonitorHandler.stopped", side_effect=[False] * iterations + [True]):
                     with patch("time.sleep"):
+                        if enabled_operations is not None:
```

Comment:
This mock was breaking the mock below (they mock the same method, and one is nested within the other)

Applying this mock conditionally (the test below doesn't need this mock)